packages/compiler/test/checker/using.test.ts

```diff
@@ -241,6 +241,43 @@ describe("compiler: using statements", () => {
     ]);
   });
 
+  it("report ambiguous diagnostics when symbol exists in using namespace and global namespace", async () => {
+    testHost.addTypeSpecFile(
+      "main.tsp",
+      `
+      import "./a.tsp";
+      import "./b.tsp";
+      `
+    );
+    testHost.addTypeSpecFile(
+      "a.tsp",
+      `
+      model M {}
+
+      using B;
+      
+      model Q extends M {}
+      `
+    );
+    testHost.addTypeSpecFile(
+      "b.tsp",
+      `
+      namespace B {
+        model M { }
+      }
+      `
+    );
+
+    const diagnostics = await testHost.diagnose("./", { nostdlib: true });
+    expectDiagnostics(diagnostics, [
+      {
+        code: "ambiguous-symbol",
+        message:
+          '"M" is an ambiguous name between global.M, B.M. Try using fully qualified name instead: global.M, B.M',
```

Comment:
I'm not sure how to phrase it, but I wonder if we should encourage putting types in global you control into a namespace. I think that's generally the better fix if you control things than to have to use `global.` and leave humans confused about the conflicting types. I would say generally, just as in C#, it's best to avoid the global namespace entirely unless you're just experimenting rather than writing something for production.

May something like "or moving global.M to a namespace"? I'm not sure this is needed, just thinking out loud.

Reply:
My 2 cents would be I think this diagnostic is probably fine as-is and we should consider adding a warning diagnostic that essentially says "Don't put things in the global namespace". I think they are two separate things.